webpack/components/extensions/HostDetails/Tabs/__tests__/packagesTab.test.js

```diff
@@ -133,10 +160,522 @@ test('Can filter by package status', async (done) => {
+
+    const selectAllCheckbox = getByLabelText('Select all');
+    selectAllCheckbox.click();
+    expect(queryByText(`${mockPackages.total} selected`)).toBeInTheDocument();
+
+    expect(getByLabelText('Select row 0').checked).toEqual(true);
+    getByLabelText('Select row 0').click(); // de select
+
+    expect(queryByText(`${mockPackages.total - 1} selected`)).toBeInTheDocument();
+    expect(getByLabelText('Select row 0').checked).toEqual(false);
+
+    // goto next page
+    getAllByLabelText('Go to next page')[0].click();
+    await patientlyWaitFor(() => expect(getAllByText('Upgradable')[0]).toBeInTheDocument());
+
+    expect(getByLabelText('Select row 0').checked).toEqual(true);
+    getByLabelText('Select row 0').click(); // de select
+
+    expect(queryByText(`${mockPackages.total - 2} selected`)).toBeInTheDocument();
+    expect(getByLabelText('Select row 0').checked).toEqual(false);
+
+    // goto previous page
+    getAllByLabelText('Go to previous page')[0].click();
+    await patientlyWaitFor(() => expect(getAllByText('Upgradable')[0]).toBeInTheDocument());
+
+    expect(getByLabelText('Select row 0').checked).toEqual(false);
+    expect(getByLabelText('Select row 1').checked).toEqual(true);
+    getByLabelText('Select row 1').click(); // de select
+
+    expect(queryByText(`${mockPackages.total - 3} selected`)).toBeInTheDocument();
+    expect(getByLabelText('Select row 1').checked).toEqual(false);
+
+    assertNockRequest(autocompleteScope);
+    assertNockRequest(scope, done); // Pass jest callback to confirm test is done
+    assertNockRequest(scope2, done);
+    assertNockRequest(scope3, done);
+  });
+
+  test('Can select page and select only items on the page', async (done) => {
+    const autocompleteScope = mockForemanAutocomplete(nockInstance, autocompleteUrl);
+    const mockPackages = makeMockPackages({});
+    const scope = nockInstance
+      .get(hostPackages)
+      .query(defaultQuery)
+      .reply(200, mockPackages);
+
+    const {
+      getByText,
+      queryByText,
+      getAllByText,
+      getByLabelText,
+      getAllByLabelText,
+    } = renderWithRedux(<PackagesTab />, renderOptions());
+
+    // Assert that the packages are now showing on the screen, but wait for them to appear.
+    await patientlyWaitFor(() => expect(getAllByText('Up-to date')[0]).toBeInTheDocument());
+
+    const selectDropDown = getAllByLabelText('Select')[0];
+    selectDropDown.click();
+
+    const selectPage = getByText('Select page (20)');
+    expect(selectPage).toBeInTheDocument();
+    selectPage.click();
+
+    expect(queryByText('20 selected')).toBeInTheDocument();
+    expect(getByLabelText('Select row 0').checked).toEqual(true);
+
+    assertNockRequest(autocompleteScope);
+    assertNockRequest(scope, done); // Pass jest callback to confirm test is done
+  });
+});
+
+test('Can upgrade a package via remote execution', async (done) => {
```

Comment:
This is where the Select All-related tests end, so this and subsequent tests should be outside the describe block.

Reply:
Right. That is the case.

Reply:
Oops, missed that second `});` :smile: